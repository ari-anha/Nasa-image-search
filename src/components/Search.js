import React, { useState } from "react";
import PropTypes from "prop-types";
import { GrFormSearch } from "react-icons/gr";
import getImages from "../requests/getImages";
import "../styles/Search.css";

const Search = ({ setSearchResults }) => {
  const [value, setValue] = useState("");

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));

    // setValue(""); ???
  };

  return (
    <>
      <form className="searchForm" onSubmit={handleSubmit}>
        <input
          className="searchInput"
          type="text"
          onChange={(event) => setValue(event.target.value)}
        />
        <button className="submitButton" type="submit">
          <GrFormSearch />
        </button>
      </form>
    </>
  );
};

Search.propTypes = {
  setSearchResults: PropTypes.array,
};

export default Search;
