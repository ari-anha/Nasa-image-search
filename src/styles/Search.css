form {
  display: flex;
  flex-direction: row;
}

.searchInput {
  margin: auto;
  width: 60%;
  height: 40px;
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
}
.submitButton {
  width: 40px;
  height: 36px;
  border: 1px solid #00b4cc;
  background: #00b4cc;
  text-align: center;
  color: #fff;
  border-radius: 0 5px 5px 0;
  cursor: pointer;
  font-size: 20px;
}

/*media query start*/
@media screen and (min-width: 410px) {
  .searchInput {
    width: 53%;
  }
}
@media screen and (min-width: 500px) {
  .searchInput {
    width: 40%;
  }
}

@media screen and (min-width: 800px) {
  .searchInput {
    width: 20%;
  }
}
/*media query end*/
