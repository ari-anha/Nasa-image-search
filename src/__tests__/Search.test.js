import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });
});
