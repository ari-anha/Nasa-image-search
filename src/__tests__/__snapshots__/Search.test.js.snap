// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="searchForm"
  >
    <input
      class="searchInput"
      type="text"
    />
    <button
      class="submitButton"
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.8,13.8 L18,18 L13.8,13.8 Z M10.5,15 C12.9852814,15 15,12.9852814 15,10.5 C15,8.01471863 12.9852814,6 10.5,6 C8.01471863,6 6,8.01471863 6,10.5 C6,12.9852814 8.01471863,15 10.5,15 Z"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </button>
  </form>
</DocumentFragment>
`;
